packages/xo-web/src/xo-app/proxies/index.js

```diff
@@ -282,6 +285,27 @@ const Proxies = decorate([
       async deployProxy({ fetchProxyUpgrades }, proxy) {
         return fetchProxyUpgrades([await deployProxy(proxy)])
       },
+      async connectProxy() {
+        const connectProxyInfo = await confirm({
+          body: <ConnectProxyModal />,
+          icon: 'connect',
+          title: _('connectProxy'),
+        })
+
+        const proxyId = await connectProxyAppliance(connectProxyInfo)
+        if (!(await isProxyWorking(proxyId))) {
```

Comment:
If this call throws, the user will think that something went wrong with registering the proxy. I'd say we should add a `.catch(() => false)` and show the second modal as if the proxy was unhealthy, what do you think?